### ISSUE TITLE
aws::ec2::networkinterface E1041 bug

### ISSUE THREAD

**Issue by whoDoneItAgain** [aws-cloudformation/cfn-lint#3964]
### CloudFormation Lint Version

1.25.1

### What operating system are you using?

windows

### Describe the bug

[cfn-lint] E1041: {'Ref': 'EniSg'} does not match destination format of 'AWS::EC2::SecurityGroup.Id'

the custom function finds the SG Id and returns it for a !Ref in other resources. E1041 is picking it up as a format error

### Expected behavior

E1041 ignores the format error when a !Ref is to a Custom resource. I've confirmed that using a !Sub '${EniSg}' doesn't produce the error

### Reproduction template

```
Resources:
  EniSg:
    Type: Custom::SgFinder
    Properties:
      ServiceToken: test123
      SgIdentifier: Sg1

  Ec2NodeEni:
    Type: AWS::EC2::NetworkInterface
    Properties:
      Description: networkinterface
      GroupSet:
        - !Ref EniSg
      SubnetId: subnet-12345678
```

**Comment by kddejong** [comment]
I'm trying to replicate this.  What is the payload you are sending back from the custom resource?  Are you using a custom resource helper like `cfnresponse` or the `custom-resource-helper`?

**Comment by whoDoneItAgain** [comment]
Yes. Im using crnresource to get data out; either via ref or getatt.

**Comment by kddejong** [comment]
Fixed and going to have a release shortly.  Thanks for the feedback.
